src/random_ray/random_ray_simulation.cpp

```diff
@@ -111,47 +111,84 @@ void validate_random_ray_inputs()
     }
   }
 
-  // Validate solver mode
-  ///////////////////////////////////////////////////////////////////
-  if (settings::run_mode == RunMode::FIXED_SOURCE) {
-    fatal_error(
-      "Invalid run mode. Fixed source not yet supported in random ray mode.");
-  }
-
   // Validate ray source
   ///////////////////////////////////////////////////////////////////
 
   // Check for independent source
```

Comment:
Hm yes, it is good you do the checks here. But I think the checks I'm suggesting adding are not to protect from any programming error here, but rather an error that might be committed in the future.

Reply:
In response to the other comments on checking for null-ness, as you point out here they are all validated up front in the `void validate_random_ray_inputs()` function. Personally, I think it makes sense to validate inputs once up front rather than repeating the same validation many times throughout the codebase -- best not to repeat the same code many times (i.e., the DRY axiom of programming).

Reply:
Would you consider changing the dynamic casts to be to const references then? This doesn't protect if sources were added through the C API as far as I can tell.

Reply:
Eh nevermind come to think of it. This is easy to debug even if null pointers were returned.